src/boards/rainbow_esp.cpp

```diff
@@ -454,7 +506,7 @@ uint8 BrokeStudioFirmware::getFreeFileId(uint8 path) const {
 		return NOT_FOUND;
 	}
 	std::array<bool, NUM_FILES> const& existing_files = this->file_exists.at(path);
-	for (size_t i = 0; i < existing_files.size(); ++i) {
+	for (uint8 i = 0; i < static_cast<uint8>(existing_files.size()); ++i) {
```

Comment:
Better keep `size_t` for `i`'s type here. It is used to index an std::array which use `size_t` internally. Forcing `uint8` you just force two more cast by iteration: the one you wrote in the loop's condition and an implicite one when writing `existing_files[i]` as array's operator `[]` takes a `size_t` parameter.

Reply:
I made this change to make the compiler happy and fix a warning message, but it worked fine before so ok, I'll switch back.